test/GPU/linalg-to-gpu-wmma.mlir

```diff
@@ -0,0 +1,141 @@
+// RUN: tpp-opt %s -linalg-to-gpu=wmma -split-input-file | FileCheck %s
+
+func.func @matmul(%arg0: memref<16x16xf16>,
+                 %arg1: memref<16x16xf16>,
+                 %arg2: memref<16x16xf16>) {
+  linalg.matmul ins(%arg0, %arg1 : memref<16x16xf16>, memref<16x16xf16>)
+                outs(%arg2 : memref<16x16xf16>)
+  return
+}
+
+// CHECK-LABEL: func.func @matmul(
+// CHECK-SAME:  %[[A:.+]]: memref<16x16xf16>, %[[B:.+]]: memref<16x16xf16>, %[[C:.+]]: memref<16x16xf16>
+// CHECK-DAG:     %[[subgroup_size:.+]] = arith.constant 32 : index
+// CHECK-DAG:     %[[one:.+]] = arith.constant 1 : index
+// CHECK:         scf.parallel {{.*}}to (%[[one]], %[[one]])
```

Comment:
Can you please clarify why we create an outer loops with a single iteration? Is it not canonicalized away?

Reply:
Currently the kernel outlining i.e., creating GPU kernels and function launch, is modeled using nested parallel loops - outer loop models blocks, inner loop models threads.

In case of larger workloads, tiling provides me with the outer loop as GPU blocks model number of matmul blocks. Also, normally running standard eltwise kernel with only blocks and a thread/subgroup size of 1 is fine.
However, to use WMMA operation, kernel must execute as a wrap/subgroup of 32 threads. So, I need this dummy wrapper to enforce kernel outlining with 32 threads specifically for WMMA in case of small (test) workloads like this one.

Because I can run linalg to parallel loops conversion back to back with GPU parallel loop mapping (and kernel outlining), it is a good enough "hack" for now.

But indeed, it will get canonicalized away which a bit of limitation on how to properly model kernel with `blocks (1, 1, 1) threads(X, Y, Z)`. Addressing that would require better kernel mapping/outlining which is a problem for the future.